webapp/src/server/api/routers/saving.ts

```diff
@@ -0,0 +1,91 @@
+import { createTRPCRouter, protectedProcedure } from "~/server/api/trpc";
+import { PartnerIncluded } from "./partner";
+import { OfferIncluded } from "./offer";
+import { z } from "zod";
+import { Coupon } from "~/payload/payload-types";
+
+export interface CouponExtanded extends Coupon {
+  savingAmount: number | null;
+  offer: OfferIncluded;
+}
+
+export const savingRouter = createTRPCRouter({
+  getByUserId: protectedProcedure
+    .input(
+      z.object({
+        userId: z.number(),
+      })
+    )
+    .query(async ({ ctx, input }) => {
+      const { userId } = input;
+      const userCouponsDocs = await ctx.payload.find({
+        collection: "coupons",
+        depth: 5,
+        where: {
+          user: {
+            equals: userId,
+          },
+        },
+        sort: "-usedAt",
+      });
+
+      let userCoupons = userCouponsDocs.docs as CouponExtanded[];
+
+      const userSavings = await ctx.payload.find({
+        collection: "savings",
+        depth: 0,
+        where: {
+          coupon: {
+            in: userCoupons.map((coupon) => coupon.id),
+          },
+        },
+      });
+
+      userCoupons = userCoupons
+        .filter(
+          (coupon) => !!userSavings.docs.find((s) => s.coupon === coupon.id)
+        )
```

Comment:
remove to get all the coupons, even if the economy is not created yet

Reply:
Solved => bff3c8c